src/djimatrice_drone.py

```diff
@@ -0,0 +1,235 @@
+        except:
+            result = {"success":False, "message":"Mission failed to start"}
+        # self.start_mission_callback(result)
+        # TODO: Upon failure, revert back to original setting
+        return result
+
+    def start_mission_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
+        
+    def stop_mission(self):
+        try:
+            print("Attempting to stop drone mission...")
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/mission_waypoint_action', 'dji_sdk/MissionWpAction')
+            request = roslibpy.ServiceRequest({"action": Drone.WaypointActions.STOP})
+
+            print('Calling mission_waypoint_action stop service...')
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Mission failed to stop"}
+        self.stop_mission_callback(result)
+        # TODO: Upon failure, revert back to original setting
+        return result
+
+    def stop_mission_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
+
+    def pause_mission(self):
+        try:
+            print("Attempting to pause drone mission...")
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/mission_waypoint_action', 'dji_sdk/MissionWpAction')
+            request = roslibpy.ServiceRequest({"action": Drone.WaypointActions.PAUSE})
+
+            print('Calling mission_waypoint_action pause service...')
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Mission failed to pause"}
+        # TODO: Upon failure, revert back to original setting
+        return result
+
+    def pause_mission_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
+
+    def resume_mission(self):
+        try:
+            print("Attempting to resume drone mission...")
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/mission_waypoint_action', 'dji_sdk/MissionWpAction')
+            request = roslibpy.ServiceRequest({"action": Drone.WaypointActions.RESUME})
+
+            print('Calling mission_waypoint_action resume service...')
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Mission failed to resume"}
+        # TODO: Upon failure, revert back to original setting
+        return result
+    
+    def resume_mission_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
```

Comment:
Wherever we do result["success"] in this file, for safety we should replace it with result.get("success", False) just in case the result dictionary that's passed in doesn't have a "success" key.